filter/expand_json_test.go

```diff
@@ -0,0 +1,195 @@
+package filter
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/AdRoll/baker"
+)
+
+func TestExpandJSON(t *testing.T) {
+	tests := []struct {
+		name      string
+		json      string
+		field     map[string]string
+		source    string
+		trueFalse []string
+		want      map[string]string
+		wantErr   bool
+	}{
+		{
+			name: "string test1",
+			json: `{"j1": "value1", "j2": "value 2", "j3": "1,[2],3,4{"}`,
+			want: map[string]string{"f1": "value1", "f2": "value 2", "f3": "1,[2],3,4{"},
+		},
+		{
+			name: "string test2",
+			json: `{"junk1": 234, "junk2": {"a":"hello","b":[true,2]}, "j1": "text"}`,
+			want: map[string]string{"f1": "text"},
+		},
+		{
+			name: "test numbers",
+			json: `{"j1": 12, "j2": 0.15, "j3": 15.12312312312}`,
+			want: map[string]string{"f1": "12", "f2": "0.15", "f3": "15.12312312312"},
+		},
+		{
+			name: "test boolean 1",
+			json: `{"j1": true, "j2": false}`,
+			want: map[string]string{"f1": "true", "f2": "false"},
+		},
+		{
+			name:      "test boolean 2",
+			json:      `{"j1": true, "j2": false}`,
+			want:      map[string]string{"f1": "t", "f2": "f"},
+			trueFalse: []string{"t", "f"},
+		},
+		{
+			name: "other json types",
+			json: `{"j1": null, "j2": [1,2,3,4], "j3": {"j1":"a", "j2":false}}`,
+			want: map[string]string{"f1": "", "f2": "[1,2,3,4]", "f3": `{"j1":"a","j2":false}`},
```

Comment:
nit: wouldn't it be a bit more readable if this was like:
```suggestion
			want: map[string]string{
				"f1": "",
				"f2": "[1,2,3,4]",
				"f3": `{"j1":"a","j2":false}`,
			},
```